views/home.ejs

```diff
@@ -1,63 +1,174 @@
-            window.location.href = "/profile";
-        });}
-
-        if(events){
-            events.addEventListener('click',() =>{
-                window.location.href="/events"
-       
-            });
-
-     }
-    
-
-    </script>
+        <div class="navbar">
+            <p>Welcome, <%= user.username %>!</p>
+            <div class="dropdown">
+                <button class="dropbtn">Menu</button>
+                <div class="dropdown-content">
+                    <button id="logout">Logout</button>
+                    <button id="profile">Profile</button>
+                    <button id="events">Events</button>
+                    <% if (user.role=='admin' ) { %>
+                        <button id="dashboard">Analytics</button>
+                        <% } %>
+                </div>
+            </div>
+        </div>
+        <% } else { %>
+            <% if (message) { %>
+                <p>
+                    <%= message %>
+                </p>
+                <% } %>
+                    <p>Welcome to the home page</p>
+                    <div class="navbar">
+                        <div class="dropdown">
+                            <button class="dropbtn">Menu</button>
+                            <div class="dropdown-content">
+                                <button id="signUp">Sign Up</button>
+                                <button id="signIn">Login</button>
+                            </div>
+                        </div>
+                    </div>
+                    <% } %>
+
+                        <p>Explore the Event Management System for seamless event planning and organization.</p>
+
+                        <script>
+                            //button click event for sign up, sign in, and navigation
+                            const signUpButton = document.getElementById('signUp');
+                            const signInButton = document.getElementById('signIn');
+                            const logoutButton = document.getElementById('logout');
+                            const profileButton = document.getElementById('profile');
+                            const events = document.getElementById('events');
+                            const Analytics = document.getElementById('dashboard');
+
+                            if (signUpButton) {
+                                signUpButton.addEventListener('click', () => {
+                                    window.location.href = "/register";
+                                });
+                            }
+
+                            if (signInButton) {
+                                signInButton.addEventListener('click', () => {
+                                    window.location.href = "/login";
+                                });
+                            }
+
+                            if (logoutButton) {
+                                logoutButton.addEventListener('click', () => {
```

Comment:
I see possibility of making this code reusable. 

Taking `logoutButton` and `/logout` as arguments for a function.